apis/Google.Storage.V1/Google.Storage.V1/UpdateBucketOptions.cs

```diff
@@ -51,20 +52,38 @@ public sealed class UpdateBucketOptions
         /// </summary>
         public PredefinedObjectAcl? PredefinedDefaultObjectAcl { get; set; }
 
-        internal void ModifyRequest(UpdateRequest request)
+        /// <summary>
+        /// If set to true, no other preconditions must be set, and
+        /// the local metageneration of the bucket being updated is not used
+        /// to create a precondition.
+        /// </summary>
+        public bool? ForceNoPreconditions { get; set; }
+
+        internal void ModifyRequest(UpdateRequest request, Bucket bucket)
         {
             if (IfMetagenerationMatch != null && IfMetagenerationNotMatch != null)
             {
                 throw new ArgumentException($"Cannot specify {nameof(IfMetagenerationMatch)} and {nameof(IfMetagenerationNotMatch)} in the same options", "options");
             }
+            if (ForceNoPreconditions == true && (IfMetagenerationMatch != null || IfMetagenerationNotMatch != null))
```

Comment:
Seems like remembering to keep this disjunction and the similar conjunction below in sync is a minor maintenance risk. Perhaps a private method or property such as HasAnyPreconditions. Same for UpdateObjectOptions.